tests/framework/utils/k8s_helper.go

```diff
@@ -668,6 +730,28 @@ func (k8sh *K8sHelper) IsPodInExpectedState(podNamePattern string, namespace str
 //CheckPodCountAndState returns true if expected number of pods with matching name are found and are in expected state
 func (k8sh *K8sHelper) CheckPodCountAndState(podName string, namespace string, minExpected int, expectedPhase string) bool {
 	listOpts := metav1.ListOptions{LabelSelector: "app=" + podName}
+	podCountCheck := false
+	actualPodCount := 0
+	inc := 0
+	for inc < RetryLoop {
+		podList, err := k8sh.Clientset.CoreV1().Pods(namespace).List(listOpts)
+		if err != nil {
+			logger.Errorf("Cannot get logs for app : %v in namespace %v, err: %v", podName, namespace, err)
+			return false
+		}
+		actualPodCount = len(podList.Items)
+		if actualPodCount >= minExpected {
+			podCountCheck = true
+			break
+		}
+
+		inc++
+		time.Sleep(RetryInterval * time.Second)
+	}
+	if !podCountCheck {
+		logger.Errorf("Expecting %d number of pods for %s app, found %d ", minExpected, podName, actualPodCount)
+		return false
+	}
```

Comment:
can we remove lines 756-765 now? Looks like the new `for` loop already checks for the min expected